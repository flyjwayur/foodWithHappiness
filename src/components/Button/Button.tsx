/// <reference path='../../index.d.ts'/>
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import theme from '../../theme/theme';
import { mediaQuerySizes } from '../../theme/mediaQuery';
// @ts-ignore
import sortIcon from '../../assets/sortIcon.png';

const MainButton = styled.button`
  width: max-content;
  text-align: center;
  padding: 1rem 1.8rem;
  font-size: 1.2rem;
  letter-spacing: 2px;
  border: 2px solid ${theme.secondary};
  color: ${theme.lightText};
  background-color: ${theme.primary};
  border-radius: 5px;
  font-weight: 700;
  box-shadow: 0 0 8px 0 rgba(32, 33, 37, 0.12);
  cursor: pointer;
  transition: all 0.2s;
  display: flex;
  align-items: center;
  transition: all ease-in-out 0.5s;

  @media (max-width: ${mediaQuerySizes.tabletPortrait}) {
    padding: 0.8rem 1.5rem;
    font-size: 1rem;
  }

  @media (max-width: ${mediaQuerySizes.phone}) {
    font-size: 1rem;
  }

  &:hover {
    background-color: ${theme.lightPrimary};
  }
`;

const SortButtonIcon = styled.img`
  width: 1.7rem;
  height: 1.7rem;
  padding-right: 0.8rem;
`;

const Button: FunctionComponent<{
  label: string;
  onClickFn: () => void;
}> = ({ label, onClickFn }) => {
  return (
    <MainButton onClick={onClickFn}>
      <SortButtonIcon src={sortIcon} />
      {label}
    </MainButton>
  );
};

export default Button;
