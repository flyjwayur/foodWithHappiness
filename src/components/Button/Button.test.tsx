import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Button from './Button';
import 'jest-styled-components';

afterEach(cleanup);

describe('<Button />', () => {
  test('Click button to sort restaurant', async () => {
    // Arrange
    const handleSort = jest.fn();
    const label = 'Sort Restaurants';
    const ascending = true;
    const { getByTestId, getByText, findByTestId } = render(
      <Button label={label} onClickFn={handleSort} ascending={ascending} />
    );

    const button = await findByTestId('sort-button');

    // Act
    fireEvent.click(getByText('Sort Restaurants'));

    // Assert
    expect(handleSort).toHaveBeenCalledTimes(1);
    expect(getByTestId('sort-button')).toHaveTextContent('Sort Restaurants');
    expect(button).toMatchSnapshot();
  });
});
