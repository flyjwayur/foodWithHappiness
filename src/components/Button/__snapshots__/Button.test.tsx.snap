// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Click button to sort restaurant 1`] = `
.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  text-align: center;
  padding: 1rem 1.8rem;
  font-size: 1.2rem;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  border: 2px solid #f8e9a1;
  color: #ffffff;
  background-color: #009de0;
  border-radius: 5px;
  font-weight: 700;
  box-shadow: 0 0 8px 0 rgba(32,33,37,0.12);
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all ease-in-out 0.5s;
  transition: all ease-in-out 0.5s;
}

.c0:hover {
  background-color: #45b0de;
}

.c1 {
  width: 1.7rem;
  height: 1.7rem;
  padding-right: 0.8rem;
}

@media (max-width:900px) {
  .c0 {
    padding: 0.8rem 1.5rem;
    font-size: 1rem;
  }
}

@media (max-width:600px) {
  .c0 {
    font-size: 1rem;
  }
}

<button
  class="c0"
  data-testid="sort-button"
>
  <img
    class="c1"
    src=""
  />
  Sort Restaurants
</button>
`;
